logging.go

```diff
@@ -2,16 +2,51 @@ package main
 
 import (
 	"io"
+	"os"
+	"path"
 	"strings"
 
 	log "github.com/Sirupsen/logrus"
 )
 
-func initLog(logLevel string, out io.Writer) {
+var logWriter io.WriteCloser
+
+func reloadLog(logdir string, logFileName string, logLevel string) {
```

Comment:
I think you can combine `reloadLog` and `initLog` in the same function `reloadLog`. I see a repetitive pattern in calls to `initLog(logLevel, os.Stderr)` after trying to open log file.